dev/conbench_envs/README.md

```diff
@@ -0,0 +1,30 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+## Benchmark Builds Env
+This directory contains 
+- [benchmarks.env](../../dev/conbench_envs/benchmarks.env) - list of env vars used for building Arrow C++/Python/R/Java/JavaScript and running benchmarks using [conbench](https://ursalabs.org/blog/announcing-conbench/)
+- [utils.sh](../../dev/conbench_envs/utils.sh) - utils used by benchmark builds for creating conda env with Arrow C++/Python/R/Java/JavaScript built from source
```

Comment:
I'm confused here because I don't see any usage of the functions in this script within arrow. If this is providing hooks for later use by conbench, could you include a reference to the relevant conbench issue which will add that usage? If util.sh is only meant to be illustrative for users who want to use `benchmarks.env` to produce an arrow build, could you clarify that intent here?

Reply:
@bkietz Thank you for the review. I added comments that I hope will make it clear why Arrow has`utils.sh`.

Note that I was not too confident about adding `utils.sh` (hooks used by benchmark builds managed in private repo) to this PR. I think it will help with benchmark builds/Arrow code compatibility issues we ran a few times. I added it to this PR because I wanted to see what Arrow contributors think about or if they have better solutions in mind.

If you think `utils.sh` is a not a good solution or it should not have been handled in this PR (which is only for `benchmarks.env`), I can remove it completely or handle in different PR. Please let me know.

Reply:
To be clear, I think `util.sh` has merit here; IIUC it represents a one button "build arrow for benchmarking" script which is definitely useful- and not just to conbench. I think consumption of `utils.sh` by other code (whether private or public) is not an issue here, so long as we clarify the interface we're providing and make it clear how a consumer can be wired up.

For example if the hypothetical project cenbonch happened to have similar requirements to conbench and wanted to reuse `benchmark.env` and `util.sh` it wouldn't be immediately obvious how. I'm *guessing* that the only entry point which conbench requires is `create_conda_env_with_arrow_python`, so a cenbonch developer could safely replace `build_arrow_cpp, build_arrow_r, ...` with a monolithic `build_arrow` if that were useful for some reason and `@ursabot benchmark please` would continue to work. However without an explicit statement of ground rules for what should and shouldn't be in `util.sh` I'd have to infer it through trial and error

Reply:
@bkietz I will define ground rules. Give me 24 hours.

Reply:
@bkietz Done. Please let me know if README.md makes sense to you about hooks are used by private repo and how they can be used by others. Will you please also approve CI workflow? Thank you!

Reply:
@bkietz Thank you so much for the review.

I also added this comment. I hope it is ok.
`- benchmark builds are run using `@ursabot please benchmark` PR comment to confirm that function definition updates do not break benchmark builds.`. 

https://github.com/apache/arrow/pull/10667/files#diff-04649ad45f96fd68437c83c30ad9dd25df8f6cf865d2a799b860eb7a159e11fcR47